### PR TITLE
Adds InstanceBufferObject and InstanceBufferObjectSubData to enable instanced rendering

### PR DESCRIPTION
**Pull request by mrdlink**
Adds InstanceBufferObject and InstanceBufferObjectSubData to enable instanced rendering.
Requires GLES3+ context. Must be used with VertexBufferObjectWithVAO.
Offers the same interface as VertexBufferObject and VertexBufferObjectSubData.

Both classes are tested for indexed and non-indexed rendering. There may be a lack of configuring the glVertexAttribDivisor, because it is set to 1 for each VertexAttribute and can not be changed.

### CODE REVIEW

**Comment by intrigus** [comment]
Do you know if this would work with WebGL2?
(I know that libgdx currently doesn't have WebGL2 support iirc)

**Comment by mrdlink** [comment]
If you can't create a WebGL2 context with libgdx, then it won't work. Instanced rendering is not available in WebGL1.0/GLES2.

**Comment by intrigus** [comment]
But It would work if libgdx supported webgl2 right? 
Webgl2 is like GLES3 but there are always corner cases that's why I'm asking.

**Comment by mrdlink** [comment]
Yes it would work with WebGL2. The [API](https://developer.mozilla.org/en-US/docs/Web/API/WebGL2RenderingContext) clearly has all functions included for instanced rendering.

File: gdx/build.gradle [hunk 1]
```diff
@@ -29,6 +29,7 @@ compileJava {
 }
 
 dependencies {
+    compileOnly "org.joml:joml:1.9.8"
```
Comment (Tom-Ski) [inline gdx/build.gradle]: Left in by mistake?

**Comment by Tom-Ski** [comment]
Can you provide a test for this please?

File: gdx/build.gradle (hunk 1 above)
Comment (mrdlink) [reply gdx/build.gradle]: Sorry, yes. I tried to experiment with JOML. I'll delete the dependency.

**Comment by mrdlink** [comment]
Should I just create a JUnit test, or in what form should that test be?

Tomski <notifications@github.com> schrieb am Di., 27. März 2018, 22:11:

**Comment by Tom-Ski** [comment]
Thanks for the contributions! This has been updated slightly and manually merged.

**Pull request closed without merging**